src/server/session.ts

```diff
@@ -1868,9 +1868,9 @@ namespace ts.server {
 
         private getFileAndLanguageServiceForSyntacticOperation(args: protocol.FileRequestArgs) {
             // Since this is syntactic operation, there should always be project for the file
-            // we wouldnt have to ensure project but rather throw if we dont get project
+            // throw if we dont get project
             const file = toNormalizedPath(args.file);
-            const project = this.getProject(args.projectFileName) || this.projectService.tryGetDefaultProjectForFile(file);
+            const project = this.getProject(args.projectFileName) || this.projectService.ensureDefaultProjectForFile(file);

```

Comment:
Is this going to have perf implications?

Reply:
We use it to get LanguageService To invoke the methods on .. 
There shouldnt be perf penalty as the updated function is `this.tryGetDefaultProjectForFile(fileName) || this.doEnsureDefaultProjectForFile(fileName);` so only if previously we were throwing no project found we would be doing the extra work to ensure that there is project for the script info which is intended.